nemo/collections/nlp/parts/nlp_overrides.py

```diff
@@ -381,6 +390,14 @@ def save_checkpoint(
             checkpoint['state_dict'] = OrderedDict([])
 
             self.checkpoint_io.save_checkpoint(checkpoint, ckpt_to_dir(filepath), storage_options=storage_options)
+
+            if HAVE_MODELOPT:
```

Comment:
I'm not sure I understand this, why do we save the modelopt states just because we have modelopt installed?

Reply:
All these functions will not do anything if the model has not been modified by modelopt